### ISSUE TITLE
Comparison is always true in Realtek RTL9300 PHY/SERDES driver

### ISSUE THREAD

**Issue by cpackham-atlnz** [openwrt/openwrt#15689]
### Describe the bug

I noticed this in passing. I haven't actually got a working system so I'm just raising the bug instead of attempting what should be a reasonably straight forward fix.

https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=target/linux/realtek/files-5.15/drivers/net/phy/rtl83xx-phy.c;hb=HEAD#l2425

The comparison `if (phy_mode == PHY_INTERFACE_MODE_10GBASER || PHY_INTERFACE_MODE_1000BASEX)` will always be true. I suspect that last condition should be `|| phy_mode == PHY_INTERFACE_MODE_1000BASEX)`

This probably will be flagged as an warning/error with the right compiler settings  (e.g. -Wtautological-compare).

### OpenWrt version

master

### OpenWrt release

master

### OpenWrt target/subtarget

realtek/rtl930x

### Device

realtek/rtl930x

### Image kind

Self-built image

### Steps to reproduce

N/A found through code inspection

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by bevanweiss** [comment]
@cpackham-atlnz could you check if this can be closed now? (and if so, close it..)

**Comment by cpackham-atlnz** [comment]
Yep looks like this was fixed in commit d6977ab33a ("realtek: rtl930x: sgmii support")
